client-go/cmd/auth.go

```diff
@@ -0,0 +1,212 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/deis/deis/client-go/controller/client"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// Register a user with the controller
+func Register(controller string, username string, password string, email string,
```

Comment:
this is slightly confusing. In terms of Go best practices, the `cmd` package should be a "dumb" implementation of each command, only being given an argument list which then it can pass along to the underlying client libraries. For example, in golang.org/src/cmd/go/build.go, they use a [Command structure](http://golang.org/src/cmd/go/build.go#L30), which then calls [rubBuild](http://golang.org/src/cmd/go/build.go#L261) that takes an argument list. This is also replicated amongst other projects in the community. I think it'd be better to standardize and follow the same best practices as other projects. That way each command can specify its own flagset that is independent of the root binary's flagset, among other benefits (standardization, ability for users to call cmd.Register with an argument list without it breaking across versions etc.). What do you think about taking this approach?